pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -137,6 +151,16 @@ func (rc *reconciler) reconcile() {
 	for _, attachedVolume := range rc.actualStateOfWorld.GetAttachedVolumes() {
 		if !rc.desiredStateOfWorld.VolumeExists(
 			attachedVolume.VolumeName, attachedVolume.NodeName) {
+
+			// Don't even try to start an operation if there is already one running
+			// This check must be done before we do any other checks, as otherwise the other checks
+			// may pass while at the same time the volume leaves the pending state, resulting in
+			// double detach attempts
+			if rc.attacherDetacher.IsOperationPending(attachedVolume.VolumeName, "") {
```

Comment:
In that case the volume would have been removed from the actualStateOfWorld and thus the next call SetDetachRequestTime will complain about the volume to being found and thus bail out. I could make this more explicit and add a VolumeExists check just after the IsPending check.